### PR TITLE
refactor(viewer.js): Add errorinterceptor argument to propagate error to the top level app 

### PR DESCRIPTION
**Pull request by GitanjaliChhetri**
Errors thrown in _viewer.js_ can now be propagated to the Slim Viewer App.
Changes needed for https://github.com/herrmannlab/slim/pull/134

### CODE REVIEW

**Comment by GitanjaliChhetri** [comment]
@hackermd I do not seem to have permissions to add reviewers to this PR. Please know it's ready for your review. Thanks!

File: src/viewer.js [hunk 1]
```diff
@@ -749,24 +749,28 @@ class VolumeImageViewer {
...
    * @param {number[]} [options.highlightColor=[140, 184, 198]] - Color that
    * should be used to highlight things that get selected by the user
+   * @param {number[]} [options.errorInterceptor] - An error instance to propogate 
+   * any error thrown to the top level app
    */
   constructor (options) {
     this[_options] = options
 
+    if(!('errorInterceptor' in this[_options])){
+      this[_options].errorInterceptor = error => {throw error}}
```
Comment (hackermd) [inline src/viewer.js:759]:
```suggestion
      this[_options].errorInterceptor = error => {throw error}
    }
```

File: src/viewer.js [hunk 2]
```diff
@@ -749,24 +749,28 @@ class VolumeImageViewer {
...
     this[_clients] = {}
     if (this[_options].client) {
       this[_clients].default = this[_options].client
     } else {
       if (this[_options].clientMapping == null) {
-        throw new Error(
-          'Either option "client" or option "clientMapping" must be provided.'
-        )
+        this[_options].errorInterceptor(new Error(
+          'Either option "client" or option "clientMapping" must be provided.'))
```
[the remark this answers is not in the document]
Comment (hackermd) [reply src/viewer.js:767]: Please also place closing brackets `)` on a new line.

File: src/viewer.js [hunk 3]
```diff
@@ -4842,6 +4856,9 @@ class LabelImageViewer extends _NonVolumeImageViewer {
    * should be included for correction of image colors
    */
   constructor (options) {
+    if(!('errorInterceptor' in options)){
```
Comment (hackermd) [inline src/viewer.js:4859]:
```suggestion
    if(options.errorInterceptor == null){
```

File: src/viewer.js [hunk 4]
```diff
@@ -749,24 +749,28 @@ class VolumeImageViewer {
    * the application
    * @param {number[]} [options.highlightColor=[140, 184, 198]] - Color that
    * should be used to highlight things that get selected by the user
+   * @param {number[]} [options.errorInterceptor] - An error instance to propogate
```
[the remark this answers is not in the document]
Comment (GitanjaliChhetri) [reply src/viewer.js]: Thanks for the review, @hackermd .
I have updated the documentation comments accordingly. Also, modified all error instances by assigning to a variable first.

**Comment by GitanjaliChhetri** [comment]
@hackermd I have addressed the requested changes. Please let me know if this can be merged? Thank you.

**Review by hackermd — CHANGES_REQUESTED** [review]
Thanks @GitanjaliChhetri. The changes look good. However, the `yarn test` command currently fails for me. The error report includes a lot of linting errors that were discovered by `yarn lint`.

Unfortunately, the ["unit tests" workflow](https://github.com/ImagingDataCommons/dicom-microscopy-viewer/blob/master/.github/workflows/run_unit_tests.yml) doesn't automatically get tricked for the pull request. I somehow can't add it as a required status check. cc @fedorov 
Until this issue is resolved, the checks need to be run manually locally.

**Comment by hackermd** [comment]
@GitanjaliChhetri @fedorov got the status checks working.

**Comment by fedorov** [comment]
@hackermd can you please re-review? Per @GitanjaliChhetri, all of your concerns should now be addressed.

**Review by hackermd — APPROVED** [review]

**Review by igoroctaviano — APPROVED** [review]

**Pull request merged**